src/utils/__tests__/reporting_workflow.test.js

```diff
@@ -227,4 +227,35 @@ describe('Test reporting workflow', () => {
     // G must be before I
     expect(g).toBeLessThan(i);
   });
+
+  test('execution works with offsets between actions', done => {
+    // setup a new action specifically for this test
+    EXAMPLE_ACTION_IMPL['do-something-offset'] = () => done();
+
+    const planDef = {
+      resourceType: 'PlanDefinition',
+      meta: {
+        profile: [MEDMORPH_PLANDEF_PROFILE]
+      },
+      action: [
+        buildAction({ code: 'init-counter', next: 'increment-counter' }),
+        buildAction({ code: 'increment-counter', next: 'do-something-offset' }),
+        buildAction({ code: 'do-something-offset', next: 'create-report' }),
+        CREATE_REPORT_ACTION
+      ]
+    };
+
+    const duration = {
+      unit: 's',
+      value: 1
+    };
+
+    planDef.action[0].relatedAction[0].offsetDuration = duration;
+    planDef.action[1].relatedAction[0].offsetDuration = duration;
+
+    const context = initializeContext(planDef);
+    executeWorkflow(context);
+    // the workflow will run and call done() async.
+    // if you don't believe me, change the action impl above to not call done()
```

Comment:
Hahaha, I believe you. One thing to note is the this test then delays and runs slowly. I believe the solution should be `jest.useFakeTimers()`, but I failed to get that working for my PR and adding it to this test didn't work either. Something to consider though

Reply:
Yeah that's why I intentionally made it only 2 seconds total so it's not a huge delay, but I'll look into useFakeTimers

Reply:
Can we make it like 50ms or at least something shorter?

Reply:
Updated to 50ms